Directory.Packages.props

```diff
@@ -1,38 +1,35 @@
 <Project>
   <PropertyGroup>
-	<ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
   </PropertyGroup>
-
   <ItemGroup>
-	<PackageVersion Include="CommunityToolkit.Diagnostics" Version="8.2.2" />
-	<PackageVersion Include="coverlet.collector" Version="6.0.2" />
-	<PackageVersion Include="coverlet.msbuild" Version="6.0.2" />
-	<PackageVersion Include="DocFx.App" Version="2.77.0" />
-	<PackageVersion Include="GitHubActionsTestLogger" Version="2.4.1" />
-	<PackageVersion Include="Microsoft.Bcl.HashCode" Version="1.1.1" />
-	<PackageVersion Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.4" />
-	<PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.10.0" />
-	<PackageVersion Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" Version="3.3.4" />
-	<PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.10.0" />
-	<PackageVersion Include="Microsoft.SourceLink.GitHub" Version="8.0.0" />
-	<PackageVersion Include="MinVer" Version="5.0.0" />
-	<PackageVersion Include="Scriban" Version="5.10.0" />
-	<PackageVersion Include="System.Linq.Async" Version="6.0.1" />
-	<PackageVersion Include="System.Memory" Version="4.5.5" />
-	<PackageVersion Include="ThisAssembly.Resources" Version="1.4.3" />
-	<PackageVersion Include="xunit" Version="2.9.0" />
+    <PackageVersion Include="Backport.System.Threading.Lock" Version="1.0.1" />
```

Comment:
Reviewing the code for this library, it is does not appear to be binary compatible between net8 and net9. By that I mean, a net8 consumer that references this library is referenced by a _second_ consumer which is net9. The first net8 library will look for `System.Threading.Lock` in the `Backport.System.Threading.Lock`, not find it, and throw a `MethodMissingException`. 

If I were to accept this PR, it would be on the basis of importing the code of `System.Threading.Lock` in a polyfill file directly in this library. Since at that point, the use would be internal to the library and not based on an external dependency. Then there would not be any cross net-version issues. Please consider if you would like to approach this at all.

Reply:
I don't quite understand what you mean. Anyone on anything before .NET 9.0 would use the class inside this micro library and anyone on .NET 9.0 or later would use System.Threading.Lock from the framework itself, which gives performance enhancements over locking on an object.

Reply:
Consider a `LibraryA` that is built on net8 against `Backport.System.Threading.Lock`. When it binds to `System.Threading.Lock`, it actually binds to `assembly=Backport.System.Threading.Lock;type=System.Threading.Lock`. Then a net9 application references `LibraryA`. Since it's using nuget and is on net9, it gets the net9 version of `Backport.System.Threading.Lock`. Then `LibraryA` goes to look for `assembly=Backport.System.Threading.Lock;type=System.Threading.Lock`, which doesn't exist because the net9 version of `Backport.System.Threading.Lock` doesn't have it, and it crashes the app.

If you want to improve `Backport.System.Threading.Lock`, you need to include a type-forwarder in the net9 version: 
```cs
#if NET9_0_OR_GREATER
[TypeForwardedTo(typeof(System.Threading.Lock))]
#else
// .. reimplementation for net8-
#endif
```

Reply:
Thanks for the tip! I will work on it when I get home. Otherwise LGTY?

Reply:
No, I still do not want to introduce an external dependency for the System.Threading.Lock type. If you were to introduce it as a TFM locked type into the SuperLinq assembly, that would be preferable.

Reply:
Since you've helped with your contribution, I will.

Credited you here: https://github.com/MarkCiliaVincenti/Backport.System.Threading.Lock/blob/master/CONTRIBUTORS.md

Reply:
Not quite sure why it's not building now @viceroypenguin. I tried to fix with pragma warnings but somehow they keep showing up as errors.